packages/boba/subgraphs/anchorage-dgf-sepolia/src/dispute-game-factory.ts

```
@@ -13,8 +13,55 @@ import {
   OwnershipTransferred,
   DisputeGameCreatedIndex,
 } from "../generated/schema"
-import { FaultDisputeGame } from "../generated/templates"
-import { BigInt, Bytes } from "@graphprotocol/graph-ts"
+import { FaultDisputeGame, PermissionedDisputeGame } from "../generated/templates"
+import { BigInt, Bytes, ethereum } from "@graphprotocol/graph-ts"
+
+/**
+ * Extracts l2BlockNumber from extraData bytes
+ * The l2BlockNumber is encoded in the last 8 bytes of the extraData
+ */
+function extractL2BlockNumberFromExtraData(extraData: Bytes): BigInt | null {
+  if (extraData.length < 8) {
+    return null
+  }
+
+  // Get the last 8 bytes
+  let startIndex = extraData.length - 8
+  let l2BlockNumberBytes = new Array<i32>(8)
+  for (let i = 0; i < 8; i++) {
+    l2BlockNumberBytes[i] = extraData[startIndex + i]
+  }
+
+  // Convert bytes to BigInt (big-endian)
+  let l2BlockNumber = BigInt.fromString("0")
+  for (let i = 0; i < l2BlockNumberBytes.length; i++) {
+    l2BlockNumber = l2BlockNumber.leftShift(8).plus(BigInt.fromI32(l2BlockNumberBytes[i]))
+  }
+
+  return l2BlockNumber
+}
+
+/**
+ * Extracts l2BlockNumber from transaction input using ABI decoding
+ */
+function extractL2BlockNumberFromTxInput(txInput: Bytes): BigInt | null {
+  if (txInput.length < 4) { // At least function selector
+    return null
+  }
+
+  // Skip the 4-byte function selector to get the parameters
+  let paramsData = Bytes.fromUint8Array(txInput.subarray(4))
+
+  // Decode the parameters: (uint32, bytes32, bytes)
+  let decoded = ethereum.decode("(uint32,bytes32,bytes)", paramsData)
+  if (decoded) {
+    // Extract extraData (third parameter)
+    let extraData = decoded.toTuple()[2].toBytes()
+    return extractL2BlockNumberFromExtraData(extraData)
+  }
+
+  return null
```

I'm curious about best practices here for subgraphs.  We're essentially silently failing if we don't get the parameters we expect in this path.  Is there no way to signal "we have a problem" without crashing the indexing?

---

Sure we could add a param "error" or "note" that can be null or not. Which than can be read through the subgraph.